enhancements/sig-policy/134-standalone-hub-templates/README.md

```diff
@@ -0,0 +1,185 @@
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in
+      [website](https://github.com/open-cluster-management-io/open-cluster-management-io.github.io/)
+
+## Summary
+
+To support Open Cluster Management (OCM) hub templates without the policy framework, this feature provides an additional
+`governance-standalone-hub-templating` addon which generates a user per cluster on the hub to be leveraged by the
+Configuration Policy controller for hub template resolution. The OCM hub administrator will be responsible for granting
+`list` and `watch` permissions to the resources needed by the hub templates. A sample `ConfigurationPolicy` can be a way
+to automate this for a cluster set.
+
+## Motivation
+
+When an Open Cluster Management (OCM) policy user deploys their policies with external tools such as Argo CD, the hub
+templating functionality is lost. This is often a critical feature for large scale deployments and prevents users from
+choosing an external tool if they require an alternative to the OCM policy framework.
+
+### Goals
+
+1. Support hub templates for `ConfigurationPolicy` and `OperatorPolicy` when deployed through an external tool such as
+   Argo CD.
+1. Hub templates should resolve when referenced objects are updated on the hub.
+1. Policies can still be evaluated if the managed cluster is disconnected from the hub.
+1. Permissions to the hub are per managed cluster.
+1. Provide a hub only `ConfigurationPolicy` to provide permissions to cluster sets.
+1. It must work without the rest of the policy framework on the managed cluster.
+
+### Non-Goals
+
+1. `CertificatePolicy` and Gatekeeper constraints will not be directly supported. It can be wrapped in a
+   `ConfigurationPolicy` if hub templates are needed for those.
+1. Updating a `ConfigurationPolicy` definition with hub templates while disconnected from the hub is not explicitly
+   supported. This will almost always work based on the proposed design as long as there aren't new hub template API
+   queries, but if the update happened while the hub and Configuration Policy controller were offline, it will not work.
+
+## Proposal
+
+### Design
+
+#### New governance-standalone-hub-templating Addon
+
+Currently, hub templates are resolved by the Governance Policy Propagator as part of the replicated `Policy` creation.
+In standalone mode, the Governance Policy Propagator and the Governance Policy Framework are not part of the equation.
+Because of this, the simplest and most reliable approach is to add hub template support in the Configuration Policy
+controller.
+
+A way to achieve this is to create an additional addon that is **disabled** by default called
+`governance-standalone-hub-templating`. Enabling this addon will cause the following:
+
+- The addon framework creates a user/subject represented by a certificate on the hub for each managed cluster. There
+  will be no permissions assigned to it.
```

Comment:
```suggestion
  will be no permissions assigned to it initially.
```

Reply:
We might want to consider starting it with some common permissions, if those aren't controversial? Like cluster claims and managed cluster labels?

Reply:
Ah, good point. The corresponding ManagedCluster name and labels makes sense to keep the experience consistent. The ClusterClaims are deployed on the managed cluster, so those are accessible already.

Reply:
Good point! I'll add that.